app/lanes.go

```diff
@@ -0,0 +1,108 @@
+	maxTxPerDefaultLane = 3000 // all other txs
+)
+
+var (
+	freeLaneBlockspacePercentage = math.LegacyMustNewDecFromStr("0.05")
+	defaultLaneBlockspacePercentage = math.LegacyMustNewDecFromStr("0.85")
+	mevLaneBlockspacePercentage = math.LegacyMustNewDecFromStr("0.1")
+)
+
+// CreateLanes walks through the process of creating the lanes for the block sdk. In this function
+// we create three separate lanes - MEV, Free, and Default - and then return them.
+//
+// NOTE: Application Developers should closely replicate this function in their own application.
+func CreateLanes(app *OsmosisApp, txConfig client.TxConfig) (*mevlane.MEVLane, *base.BaseLane, *base.BaseLane) {
+	// 1. Create the signer extractor. This is used to extract the expected signers from
+	// a transaction. Each lane can have a different signer extractor if needed.
+	signerAdapter := signerextraction.NewDefaultAdapter()
+
+	// 2. Create the configurations for each lane. These configurations determine how many
+	// transactions the lane can store, the maximum block space the lane can consume, and
+	// the signer extractor used to extract the expected signers from a transaction.
+	//
+	// IMPORTANT NOTE: If the block sdk module is utilized to store lanes, than the maximum
+	// block space will be replaced with what is in state / in the genesis file.
+
+	// Create a mev configuration that accepts 1000 transactions and consumes 20% of the
+	// block space.
+	mevConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   mevLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerMEVLane,
+	}
+
+	// Create a free configuration that accepts 1000 transactions and consumes 5% of the
+	// block space.
+	freeConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   freeLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerFreeLane,
+	}
+
+	// Create a default configuration that accepts 1000 transactions and consumes 60% of the
+	// block space.
+	defaultConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   defaultLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerDefaultLane,
+	}
+
+	// 3. Create the match handlers for each lane. These match handlers determine whether or not
+	// a transaction belongs in the lane.
+
+	// Create the final match handler for the mev lane.
+	factory := mevlane.NewDefaultAuctionFactory(txConfig.TxDecoder(), signerAdapter)
+	mevMatchHandler := factory.MatchHandler()
+
+	// Create the final match handler for the free lane.
+	freeMatchHandler := freelane.DefaultMatchHandler()
```

Comment:
The only one they were 100% sure of when we talked w/ them was claiming staking rewards--which I don't think is what the default match handler has.

Reply:
The FreeLane match-handler matches to (staking.Delegate, Redelegate, Unbond). I wasn't sure whether it made sense to pre-emptively implement, or point out in PR and resolve in discussion, so I did not update the MatchHandler

Reply:
Happy to update the MatchHandler accordingly tho